src/components/page-search/page-search.js

```diff
@@ -1,18 +1,29 @@
-import React from 'react';
-import SubredditForm from './subreddit-form';
+import React, { useState } from 'react';
 import useFetchPosts from '../../hooks/use-fetch-posts';
+import SearchContext from './context';
+import SubredditForm from './subreddit-form';
+import Heatmap from './heatmap';
 import * as S from './page-search.style';
-import Container from '../common/container';
 
 const SearchPage = () => {
-  const { status } = useFetchPosts();
+  const { posts, status } = useFetchPosts();
+  const [selectedPostIds, setSelectedPostIds] = useState([]);
+
+  const handleSelectedPostIds = (isAddOperation, postIds) => {
```

Comment:
This will be a lot simpler if you use a 2D or 3D array. You then don't need to store the selectedPostIds but rather a representation of the selected cell (e.g. weekday and hour)

Reply:
Agreed.